### PR TITLE
Liblapack atlas dependency optional

### PR DESCRIPTION
**Pull request by thinkpad20**
Requires #8023.

This solves an annoying issue I was facing, which is that in order to build liblapack, a relatively light-weight dependency, it was necessary to build atlas, a behemoth of a dependency. It also means that in order to build atlas with liblapack support (as is necessary to get it to take with numpy on OSX -- which will be an upcoming pull req), it was necessary to build it _twice_, once without liblapack, and then building liblapack, and then building it again. Yeesh!

Also, I consolidated the two virtually identical .nix files for liblapack into one. The only difference seemed to be version, and this makes it easier to have changes on one reflected in the other.

### CODE REVIEW

**Comment by thinkpad20** [comment]
@peti

**Comment by peti** [comment]
I like the idea of having the ability to build liblapack without atlas, but I don't like the one expression that mixes build instructions for two versions. This is really uncommon in Nixpkgs; virtually no other expressions works this way. It's hard to read, IMHO, and it's hard to maintain because every change almost always affects two versions at the same time.

**Comment by thinkpad20** [comment]
@peti there are advantages to having changes affect two versions at the same time (for example, I don't have to copy/paste the fix I made into each file). That's only a disadvantage when there's something different about the versions which necessitates splitting things up. It's just being DRY. I'm also not sure what makes it hard to read...

**Comment by thinkpad20** [comment]
@peti Case in point: in the current master, after #8023 got merged, 3.4.1 will work on OS X, while 3.5.0 will not. Literally the only difference between the two files (up until now) is the version and sha.

**Comment by peti** [comment]
As I said, almost no other package combines two versions into the same build -- even at the cost of duplicating a lot of stuff. I believe it's okay to deviate from that convention if there are strong technical reasons to do so, but I don't feel like that's the case here. Personally, I'd rather have two separate expressions for liblapack. Just my 2 cents. Others may disagree and merge this patch as-is. I'm not strongly opposed to that.

**Comment by thinkpad20** [comment]
@peti I could easily change to a two-file solution, but I just don't see the benefit to it. What's the argument for having two almost identical files, beyond the fact that almost no other package does it? DRY is one of the more important tenets of software engineering after all...

**Comment by thinkpad20** [comment]
That said I'd prioritize getting these changes in over defending this particular principle, so if it's a deal-breaker for you I can separate them and revisit this later :smile:

**Comment by peti** [comment]
I believe I have made my point. I won't merge this patch as-is, but if someone else feels differently and decides to merge this PR, then I'm totally fine with the decision.

**Comment by thinkpad20** [comment]
@peti I think it's a bit unfortunate that you're not addressing my arguments in favor of consolidation: I'm not trying to pick a fight; I just struggle to think of what advantage there is in maintaining two identical files over having one. I'd be happy to hear what those advantages are. But at the end of the day the silly thing is that a piece of beneficial functionality would be held hostage over a point such as this. So I've restored the original configuration, albeit with the cmake modification that my earlier pull request had and other functionality changes I described.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13835,7 +13835,12 @@ let
 
   jags = callPackage ../applications/science/math/jags { };
 
-  liblapack = callPackage ../development/libraries/science/math/liblapack { };
+  liblapack = callPackage ../development/libraries/science/math/liblapack { 
+    atlas = null;
+  };
```
Comment (peti) [inline pkgs/top-level/all-packages.nix]: This changes the default version of `liblapack` used in Nixpkgs so far to one that does not have ATLAS. How will that affect other software that uses liblapack, like R and haskell-hmatrix?

Comment (adnelson) [inline pkgs/top-level/all-packages.nix]: I honestly couldn't say. Perhaps I could make an `liblapackWithoutAtlas` expression and make the one with atlas the default.

**Comment by adnelson** [comment]
@peti OK, so I put all for permutations (two versions, with/out atlas) in `all-packages.nix`, and kept `liblapack` and `liblapack_3_5_0`, as defaulting to having atlas as a dependency.

**Comment by peti** [comment]
OK, very cool. Thank you for your patience!

**Pull request merged**